### DISCUSSION TITLE
Feature: Keyboard shortcuts

### DISCUSSION THREAD

**Discussion by kazad** [kazad/instacalc#106] [Announcements]
I'm experimenting with some keyboard shortcuts:

`iii`: insert row above
`ooo`: insert row below
`ddd`: delete current row
`[del][del][del]`: delete current row (if empty) - 3 quick delete presses
`ccc`: clear current row contents

The idea: you are building your calc, need a row above, and spam `i` 3 times to get a new row. You have to press the 3 keys within about a second. This helps avoid accidental triggering.

On desktop it's a lot easier to add/remove rows. My main scenario was mobile: it's hard to use `ctrl + key` to do anything special. Spamming space tends to autoreplace with `.`, hence the triple-letter combos. Deleting a bunch of rows isn't bad: just keep hitting `d`.

**Comment by chr-wn** [comment]
such a thougthful addition! would love it if ooo inserted a line above too

**Reply by kazad** [reply]
Thanks! Did you mean `iii` for above and `ooo` for below?

**Comment by chr-wn** [comment]
Oh yes, like the vim motion.

**Reply by kazad** [reply]
Yep, exactly! Just added the feature.

**Reply by chr-wn** [reply]
Oops, I meant to write that as a reply. Thanks so much! It works very well; just doesn't focus on the new line sometimes.

**Reply by kazad** [reply]
Oh, good catch! Just fixed, thanks.
